### DISCUSSION TITLE
Secondary/Approver Credentials

### DISCUSSION THREAD

**Discussion by ccopsey** [renovatebot/renovate#18086] [Suggest an Idea]
With respect to the problems posed by [mandatory pull requests reviews](https://docs.renovatebot.com/key-concepts/automerge/#required-pull-request-reviews) and the need for solutions such as [renovate-approve-2](https://github.com/apps/renovate-approve-2), would a simple and elegant solution be: to allow the specification of a secondary set of credentials?

The "primary" credentials are used (as currently) to create/update pull requests, and the "secondary" credentials (if they exists) are used to approve the pull requests?  Am I missing a glaring downside here?

As an Azure DevOps user I have browsed the code and believe that [this line](https://github.com/renovatebot/renovate/blob/e912f89fa82c5f74033ed41e9b57201efa5c43d8/lib/modules/platform/azure/index.ts#L461) would need to reference a new `azureApiGitSecondary` rather than `azureApiGit`, which in turn would require an additional [azureObjSecondary()](https://github.com/renovatebot/renovate/blob/e912f89fa82c5f74033ed41e9b57201efa5c43d8/lib/modules/platform/azure/azure-got-wrapper.ts#L22).  Aside from some extra configuration options, that appears to be the extent of it.

If agreed, I would be happy to give this a go, but I have limited experience/knowledge of other platforms.

**Comment by rarkins** [comment]
A challenge would be thinking of and handling all the edge cases. It would probably be simple to handle the simple case of "approve your own PR with secondary credentials when you create it". But then things like:
- Do you need to re-approve the PR if you force push to it to update it?
- What if someone else modifies it?
- etc.

**Reply by viceice** [reply]
i don't like that idea at all. renovate shouldn't be able to bypass protections.
